File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.js
function solution(n) {
    
    let n_count = (n.toString(2)).match(/1/g).filter(item => item !== '').length;
    
    let number = n+1;
    
    while(true){
        let count = (number.toString(2)).match(/1/g).filter(item => item !== '').length;
        
        if(count === n_count){
            break;
        }
        
        number++;
    }
    
    return number;
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        

        long n_count = Integer.bitCount(n);

        int number = n+1;
        
        while(true){
            
            long binary = Integer.bitCount(number);
            
            if(binary == n_count){
                break;
            }
            
            number++;
        }
        
        return number;
        
    }
}